### PR TITLE
Testing adjustments

### PR DESCRIPTION
**Pull request by imrehg**
Improving testing, WIP.

### CODE REVIEW

**Comment by tcbegley** [comment]
# [Codecov](https://codecov.io/gh/facultyai/faculty/pull/157?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@25f5533`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/facultyai/faculty/pull/157?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##             master     #157   +/-   ##
=========================================
  Coverage          ?   96.54%           
=========================================
  Files             ?       24           
  Lines             ?     1853           
  Branches          ?      116           
=========================================
  Hits              ?     1789           
  Misses            ?       34           
  Partials          ?       30           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/facultyai/faculty/pull/157?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/facultyai/faculty/pull/157?src=pr&el=footer). Last update [25f5533...4612d42](https://codecov.io/gh/facultyai/faculty/pull/157?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by imrehg** [comment]
Was experimenting with codecov.io
The above comment is due to adding the codecov app to this single repo. I totally don't like that it has posted with Tom's user, and would remove this app after all (even if it doesn't have any real access, just code read and comment read/write).

Code-coverage would be nice and I find useful info already when I looked it, but will try to achieve it with a less intrusive way.

**Comment by imrehg** [comment]
An alternative is actually setting a bot for this, which would be following https://docs.codecov.io/docs/team-bot but maybe it's not important for now.

**Comment by imrehg** [comment]
Now relogged in Codeconv (so report links work) and triggered a new build, if that pots with anything else, but `codecov-io` user, then indeed need to purge. As confusing as it gets.

**Comment by acroz** [comment]
I see that the code coverage has been added as a check - what causes it to fail? Reduction in code coverage?

**Comment by imrehg** [comment]
@acroz I'm happy to discuss this whole PR and what went down above in more details! It might be higher bandwidth than in writing.

The check failures above are no real failures, most of it is "early cancellation" as I've pushed new stuff before a check has been finished (maybe that part should be taken out of the travis settings). The coverage really didn't drop so far either, since no code change is present in this PR.

All of it is very experimental here, a partially misguided Sunday morning exploration. :P But hope the result will be worth it down the line.

File: .travis.yml [hunk 1]
```diff
@@ -1,36 +1,30 @@
 language: python
+python:
+  - 2.7
+  - 3.5
+  - 3.6
+  - 3.7
+  - 3.8
+  - "nightly"
```
Comment (acroz) [inline .travis.yml]: I'd be keen not to test against the nightly build as this implies we support the nightly build, which I don't think we want.

File: .travis.yml [hunk 2]
```diff
@@ -1,36 +1,30 @@
...
 cache: pip
-dist: xenial
+dist: bionic
+install:
+  - pip install tox-travis codecov
+script:
+  - tox
+after_success:
+    - codecov
```
Comment (acroz) [inline .travis.yml]: Nit: Inconsistent indentation

File: tox.ini [hunk 3]
```diff
@@ -1,14 +1,17 @@
 [tox]
-envlist = py27, py34, py35, py36, py37, py38, flake8, black, license
+envlist = py{27,34,35,36,37,38}, flake8, black, license
```
Comment (acroz) [inline tox.ini]: Might as well removed 3.4 from here too, especially since it's end of life.

File: .travis.yml [hunk 4]
```diff
@@ -1,36 +1,30 @@
...
-    - python: 3.6
-      env: TOXENV=py36
     - python: 3.7
-      env: TOXENV=py37
-    - python: 3.8
-      env: TOXENV=py38
 
-install:
-  - pip install tox
+    - python: 3.7
```
Comment (acroz) [inline .travis.yml]: I liked having the `name` fields on the linting / license checks as these make the travis test results easier to consume. Does `jobs` support that too?

File: .travis.yml (hunk 1 above)
Comment (imrehg) [reply .travis.yml]: To me this is an early warning system, just testing for it we could find issues beforehand. Could be very very very specific in the README as per what's supported, while still knowing when things break in advance. Don't think the travis config should oblige us to anything.

But if this above doesn't make sense, will remove, as I am onboard with your logic - except for the premise. If my above premise is wrong, then your point stands completely.

File: .travis.yml (hunk 4 above)
Comment (imrehg) [reply .travis.yml]: See what you mean completely! Checking it.

Comment (imrehg) [reply .travis.yml]: Yeah, previously it was jobs as well, I dropped the names by an oversight, now added all of them back.

**Comment by tcbegley** [comment]
Hey @imrehg, only just saw this, not sure why codecov was posting in my name or why my account is editing codecov reports. I use it on one of my personal repos but never configured it for any faculty repos. I don't actually have access to the settings for this repo or the Faculty organisation so I'm not quite sure how to fix it. I can't see anything obvious on the codecov website itself.

**Comment by jerrode** [comment]
@tcbegley --> jumping in here. Jerrod from Codecov. The team bot had cached as your Github username, I reset your team bot to NULL. Can you see on your next CI run / commit what comments? It *should* pick up your YAML this time.

**Comment by imrehg** [comment]
Will try to re-create this PR with the CI changes, see whether it works better, it is useful.

**Comment by acroz** [comment]
AFAICT the remaining relevant change in this PR would be adding code coverage checks? As I think we discussed when you opened this PR, the team has historically had some reluctance to introduce code coverage checks. Before investing the time to update this PR, maybe let's have a call to discuss?

**Comment by imrehg** [comment]
With recent changes and moving to GitHub Actions, this PR is no longer relevant. Closing it as such.

**Pull request closed without merging**